compiler/src/dmd/globals.d

```diff
@@ -346,6 +350,8 @@ extern (C++) struct Global
 
     extern (C++) void _init()
     {
+        global.errorSink = new ErrorSinkCompiler;
```

Comment:
This should be moved out of `globals` and into main to reduce circular dependencies